pom.xml

```diff
@@ -45,16 +49,91 @@
 
   <dependencyManagement>
     <dependencies>
+<!--      Hadoop Dependencies -->
+      <dependency>
+        <groupId>org.apache.hadoop</groupId>
+        <artifactId>hadoop-common</artifactId>
+        <version>${hadoop.version}</version>
+        <exclusions>
+          <exclusion>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>*</artifactId>
+          </exclusion>
+          <exclusion>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
      <dependency>
         <groupId>org.apache.hadoop</groupId>
         <artifactId>hadoop-mapreduce-client-common</artifactId>
         <version>${hadoop.version}</version>
       </dependency>
+
+<!--      Hive Dependencies -->
+      <dependency>
```

Comment:
We should add `provided` scope to all of these or do it in the child poms, whatever is most consistent (and works!)

Reply:
gotcha - so is the expectation that these extensions always run in an environment where hadoop/hive is already on the classpath? 

I'm wondering if with the mix between 2 & 3 if it'd be better to be more explicit w/ the dependencies

Reply:
Yes on the dependencies already being on the classpath. As for 2 vs 3, possibly, we're going to have to have a bigger think and discussion about how best to deal with it. A few things we've considered are some kinds of shims, sub-modules for different versions, different branches for different versions etc. All have their pros and (often significant) cons.

Reply:
gotcha - to circle back on the `provided` scope: It looks like it needs to be explicitly called out at the submodule level as doing it on the parent / depManagement level breaks testing.

A larger revert of that was needed but it should be good now.